File: app/src/main/java/app/endershrooms/thatcatapp/db/dao/CategoryDao.java
package app.endershrooms.thatcatapp.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import app.endershrooms.thatcatapp.model.Category;
import java.util.List;

@Dao
public interface CategoryDao {

  @Query("SELECT * FROM category ORDER BY id")
  LiveData<List<Category>> getCategories();

  @Query("SELECT * FROM category WHERE id = :id")
  LiveData<Category> getCategory(int id);

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insertCategories(List<Category> categories);
}

File: app/src/main/java/app/endershrooms/thatcatapp/db/dao/CategoryDao.kt
package app.endershrooms.thatcatapp.db.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import app.endershrooms.thatcatapp.model.Category

@Dao
interface CategoryDao {

    @Query("SELECT * FROM category ORDER BY id")
    fun getCategories(): LiveData<List<Category>>

    @Query("SELECT * FROM category WHERE id = :id")
    fun getCategory(id: Int): LiveData<Category>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertCategories(categories: List<Category>)
}
